File: shakespeare-rangmanch/shakespeare-rangmanch.py
import math

for _ in range(int(input())):
    a, b = list(map(int, input().split()))
    lg = (int)(math.pow(3, math.ceil(math.log(a) / math.log(3))))
    oneone = 2*b + 1
    if oneone >= lg:
        if (int)(math.pow(3, math.floor(math.log(oneone) / math.log(3)))) == oneone:
            print("I defy you, stars")
        else:
            print("These violent delights have violent ends")
    else:
        print("These violent delights have violent ends")

File: shakespeare-rangmanch/shakespeare-rangmanch.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0)
        {
            long a = sc.nextLong();
            long b = sc.nextLong();
            long lg = pow( 3, ceil( ( log( a )/log( 3 ) ) ) );
            if ( 2*b + 1 >= lg ) {
                long oneone = 2*b + 1;
                if ( (long)pow( 3, floor( log(oneone)/log(3) ) ) == oneone ) System.out.println("I defy you, stars");
                System.out.println("These violent delights have violent ends");
            }
            System.out.println("These violent delights have violent ends");
        }
        sc.close();
    }
}
